### ISSUE TITLE
SwiftFormat on Windows should redistribute swift runtimes that it has been tested with

### ISSUE THREAD

**Issue by vinocher-bc** [nicklockwood/SwiftFormat#1775]
SwiftFormat on Windows currently doesn't redistribute  swift runtimes that it has been tested with. This can cause it to break with other versions of swift runtimes installed on a machine. Are there plans to redistribute swift runtimes in the installer?

**Comment by nicklockwood** [comment]
@vinocher-bc since I don't use Windows it's difficult for me to know the right solution here. For Linux we ship the runtime, but it bloats the download to 20MB (zipped). @compnerd was kind enough to set up the Windows pipeline for me, so I'd guess I'd defer to him about whether it makes sense to ship the runtime there too?

**Comment by compnerd** [comment]
I think that for any platform (basically anything non-Darwin), packaging the runtime is needed as there's no ABI stability. The breakages are not really announced or tracked so you can't guarantee that the tool will work. So while it may bloat the distribution, it's the safest thing.

**Comment by tristanlabelle** [comment]
This has now been completed @nicklockwood . However it has a problematic side-effect: Installing SwiftFormat will put the SwiftFormat directory in the system %Path%, and that directory now includes Swift runtime dlls. This interferes with an installed Swift toolchain, which whose apps like the swiftc compiler depend on the system %Path% to find a matching Swift runtime. Hence the compiler might now load the runtime that SwiftFormat distributes and crash on launch.

I believe this is a problem that the Swift on Windows installer needs to fix, not SwiftFormat, as it would have happened with any other app distributing the runtime and registering itself in %Path%. However I suggest that SwiftFormat include a visible warning about this situation to not be accusing of breaking the Swift toolchain.

**Comment by nicklockwood** [comment]
@tristanlabelle this pattern of installing the swift runtime in a shared location which can then affect other apps seems problematic. Is there no way to statically include the runtime libraries in the binary like we do on Linux?

**Comment by tristanlabelle** [comment]
@nicklockwood , I agree it's problematic! We don't support statically linking the Swift runtime, and if we did I believe it would prevent an Swift exe from using a Swift dll. I filed this bug on the Swift for Windows installer so we can move the conversation there: https://github.com/swiftlang/swift-installer-scripts/issues/339 . We'll need to solve this problem for all Swift apps on Windows.

**Comment by roman-bcny** [comment]
As we now have static linking on Windows available now we should be able to move this forward.
I'll look into supporting static linking in https://github.com/nicklockwood/SwiftFormat/blob/main/.github/workflows/windows_release.yml
@nicklockwood please lmk if there are any concerns or known gotchas!

**Comment by roman-bcny** [comment]
Got a working proof of concept with static linking https://github.com/roman-bcny/SwiftFormat/pull/1 - uploads to artifacts
* ~50MB swiftformat.exe
* That is building latest main, our internal codebase is currently a little behind (0.56.1), so there are some formatting changes following updates (tweaking configs also works)
* back to no changes with the build rebased to 0.56.1

Seems like it just works!
